modules/data.land/R/get.veg.module.R

```diff
@@ -2,107 +2,96 @@
                             outfolder,
                             start_date, end_date,
                             dbparms,
-                            lat, lon, site_id, 
-                            host, overwrite){
+                            new_site, 
+                            run_start,
+                            host, machine_host,
+                            overwrite){
 
   #--------------------------------------------------------------------------------------------------#
-  # Extract/load data : this step requires DB connections can't be handled by convert.inputs
+  # Extract/load data : this step requires DB connections 
+  # can be passed to convert.inputs now because process IC locally
   
-  # which sources require load_data, temporary, need to extract from somewhereelse
-  load_sources <- c("GapMacro", "NASA_FFT_Plot_Inventory", "NASA_TE_FIA")
+  lat       <- new_site$lat
+  lon       <- new_site$lon
+  site_id   <- new_site$id
+  site_name <- new_site$name   
+  
+  ## Prepare to call convert.inputs
+  pkg  <- "PEcAn.data.land"
+  bety <- dplyr::src_postgres(dbname   = dbparms$bety$dbname, 
+                              host     = dbparms$bety$host, 
+                              user     = dbparms$bety$user, 
+                              password = dbparms$bety$password)
+  con  <- bety$con
   
   if(input_veg$source == "FIA"){
     
-    veg_info <- extract_FIA(lon, lat, start_date, gridres = 0.075, dbparms)
+    fcn <- "extract_FIA"
```

Comment:
yes that's what I had in mind, one branch for sources that have their own database (for now FIA is the only case, fcn args and name might be generalized later) and one requires using `load_data`, thought these two types would require different types of fcn args